### DISCUSSION TITLE
Pipeline editor node properties panel organization

### DISCUSSION THREAD

**Discussion by akchinSTC** [elyra-ai/elyra#3004] [Ideas]
Given the amount of interest and contributions around and in the node properties panel  (rjsf work, #2973, #3000...) thought it would be good open up a discussion how best to organize them moving forward. 
For example, we currently expose quite a few k8s parameters, but these may not interest users who are running pipelines with the local runtime. Maybe we can hide them under an `Advanced Kubernetes Options` hidden menu and only show if clicked.
Same for #2973. This PR exposes doubles the number of input fields, hiding them behind an optional button would make things much cleaner.
![image](https://user-images.githubusercontent.com/13340013/200687415-2de360e8-d125-4ef0-b5e4-81c9f9362608.png)

**Comment by ajbozarth** [comment]
After some slack discussion with @marthacryan the most straight-forward implementation for this would be to enable some category headers to be collapsable similar to the TOC widget.

**Comment by ptitzler** [comment]
Adding two more items to the wishlist, based on our discussion in the dev meeting. Assuming that there are two groups of users - 'basic' and 'advanced':
- users should be able to switch between basic and advanced display via the settings editor (basic -> render advanced properties in collapsed sections (properties are visible if the section is expanded), advanced -> render advanced properties in expanded sections)
- Users should be able to permanently hide properties if they have no intention of ever using them. (This is similar to the general community request to hide features that are not supported.)
